### PR TITLE
fix abi3 interpreter discovery on Windows

### PR DESCRIPTION
**Pull request by davidhewitt**
This is an attempt to fix https://github.com/PyO3/maturin/pull/2325#issuecomment-2506155125

(Not tested yet, I wanted to push and see how this runs on CI as well as test locally.)

### CODE REVIEW

**Comment by davidhewitt** [comment]
Looks like this is working well.

Without `generate-import-lib` enabled, setting `3.13t`:

```
~\Dev\maturin> cargo run -- build --manifest-path ../pyo3-scratch/Cargo.toml -i 3.13t
   Compiling maturin v1.7.6 (C:\Users\david\Dev\maturin)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 4.28s
     Running `target\debug\maturin.exe build --manifest-path ../pyo3-scratch/Cargo.toml -i 3.13t`
🔗 Found pyo3 bindings with abi3 support for Python ≥ 3.8
⚠️  Warning: Failed to determine python platform
💥 maturin failed
  Caused by: Need a Python interpreter to compile for Windows without PyO3's `generate-import-lib` feature
  Caused by: Python interpreter `3.13t` doesn't exist
error: process didn't exit successfully: `target\debug\maturin.exe build --manifest-path ../pyo3-scratch/Cargo.toml -i 3.13t` (exit code: 1)
```

... and the same if I enable the `generate-import-lib` feature:

```
~\Dev\maturin> cargo run -- build --manifest-path ../pyo3-scratch/Cargo.toml -i 3.13t
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.23s
     Running `target\debug\maturin.exe build --manifest-path ../pyo3-scratch/Cargo.toml -i 3.13t`
🔗 Found pyo3 bindings with abi3 support for Python ≥ 3.8
⚠️  Warning: Failed to determine python platform
🐍 Not using a specific python interpreter (automatically generating windows import library)
⚠️ Warning: CPython 3.13t does not yet support abi3 so the build artifacts will be version-specific.
   Compiling pyo3-build-config v0.23.2 (C:\Users\david\Dev\pyo3\pyo3-build-config)
   Compiling pyo3-macros-backend v0.23.2 (C:\Users\david\Dev\pyo3\pyo3-macros-backend)
   Compiling pyo3-ffi v0.23.2 (C:\Users\david\Dev\pyo3\pyo3-ffi)
   Compiling pyo3 v0.23.2 (C:\Users\david\Dev\pyo3)
   Compiling pyo3-macros v0.23.2 (C:\Users\david\Dev\pyo3\pyo3-macros)
   Compiling pyo3-scratch v0.1.0 (C:\Users\david\Dev\pyo3-scratch)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 3.76s
📦 Built wheel for CPython 3.13t to C:\Users\david\Dev\pyo3-scratch\target\wheels\pyo3_scratch-0.1.0-cp313-cp313t-win_amd64.whl
```

File: src/build_context.rs [hunk 1]
```diff
@@ -234,7 +234,7 @@ impl BuildContext {
                 if !non_abi3_interps.is_empty() {
                     let interp_names: HashSet<_> = non_abi3_interps
                         .iter()
-                        .map(|interp| format!("{}{}{}", interp, interp.major, interp.minor))
+                        .map(|interp| interp.to_string())
```
[the remark this answers is not in the document]
Comment (messense) [reply src/build_context.rs:237]: Good catch!

**Review by messense — APPROVED** [review]
Thanks!

**Comment by messense** [comment]
Tested in `nh3` and it works great!

**Comment by davidhewitt** [comment]
Great!

**Comment by davidhewitt** [comment]
Oof, good spot, I think you might be right. I'll try to circle back here and cover with some tests at the same time.

**Pull request merged**